File: QytetetJava/src/modeloqytetet/Dado.java
package modeloqytetet;
/* 
     Esta clase representa
     el dado del juego
    */
public class Dado {
    // Atributos privados    
    private int valor; // Valor del dado al tirarlo
        
    // Métodos privados
    private static final Dado instance = new Dado (); // Método dado que crea
                                                      // la única instancia de dado
 
    private Dado () {valor = 1;} // Constructor de la instancia única del dado
                                 // Valor 1 por defecto
    
    // Métodos públicos 
    public static Dado getInstance () {return instance;} // Devuelve la instancia
                                                         // única del dado
    
    public int getValor () {return valor;} // Consultor: devuelve el valor actual
                                           // del dado
    
    int tirar (){
        int valor_dado = (int) Math.ceil(Math.random() * (6));
        valor = valor_dado;
        return valor_dado;
    }
    
    // Método toString, que devuelve una cadena con los campos del dado
    @Override
    public String toString (){
        return "Dado{Valor: " + valor+ "}";
    }
    
    
}

File: QytetetRuby/pkg/QytetetRuby-0.0.1/lib/ModeloQytetet/dado.rb
#encoding: utf-8
require "singleton"
module ModeloQytetet
  
  class Dado
    
    # Clase singleton
    
    include Singleton
    attr_reader:valor
    
    # Método que genera un número entre 1-6
    
    def tirar
      random = Random.new
      @valor = random.rand(6) + 1
      return @valor
    end
    
    # toString()
    
    def to_s
      to_return = "Dado:\n Valor = #{@valor}"
      to_return
    end
  end
end
